toolchain/check/import_ref.cpp

```diff
@@ -1233,8 +1233,14 @@ static auto AddAssociatedEntities(ImportContext& context,
       auto function =
           context.import_functions().Get(function_decl->function_id);
       import_name_id = function.name_id;
+    } else if (auto import_ref =
+                   context.import_insts().TryGetAs<SemIR::AnyImportRef>(
+                       inst_id)) {
+      import_name_id =
+          context.import_entity_names().Get(import_ref->entity_name_id).name_id;
     } else {
-      CARBON_CHECK("Unhandled associated entity type");
```

Comment:
Oh that's a sneaky bug, I wonder if we can catch that at compile time.

Reply:
We can! #4628